src/graph/InsertEdgeExecutor.cpp

```diff
@@ -57,31 +56,45 @@ void InsertEdgeExecutor::execute() {
         auto dst = row->dstid();
         auto rank = row->rank();
         auto expressions = row->values();
-        std::vector<VariantType> values;
 
+        // Now default value is unsupported
+        if (expressions.size() != schema_->getNumFields() ||
+            properties_.size() != schema_->getNumFields()) {
+            LOG(ERROR) << "Input values number " << expressions.size()
+                       << ", props number " << properties_.size()
+                       << ", schema field number " << schema_->getNumFields();
+            return Status::Error("Wrong number of fields");
+        }
+
+        std::vector<VariantType> values;
         values.reserve(expressions.size());
         for (auto *expr : expressions) {
             values.emplace_back(expr->eval());
         }
 
         RowWriter writer(schema_);
+        auto schemaIndex = 0u;
         for (auto &value : values) {
-            switch (value.which()) {
-                case 0:
-                    writer << boost::get<int64_t>(value);
-                    break;
-                case 1:
-                    writer << boost::get<double>(value);
-                    break;
-                case 2:
-                    writer << boost::get<bool>(value);
-                    break;
-                case 3:
-                    writer << boost::get<std::string>(value);
-                    break;
-                default:
-                    LOG(FATAL) << "Unknown value type: " << static_cast<uint32_t>(value.which());
+            // Check field name
+            auto schemaFileName = schema_->getFieldName(schemaIndex);
```

Comment:
typo.

Reply:
We don't have to check on the property names in every iteration. You could perform a optimization, or add a comment on this as a reminder.

Reply:
why don't need?